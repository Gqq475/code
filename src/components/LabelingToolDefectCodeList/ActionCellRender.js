import React from 'react'
import { connect } from 'react-redux'
// import DeleteDialog from './DeleteDialog'
import { Button } from '@blueprintjs/core'
import {
  delLabelingToolImageDefect,
  saveLabelingToolImageDefect,
  updateLabelingToolImageDefect,
  DELETE_DEFECT_MARK_FROM_INDEX
} from 'routes/OJS/modules/OJS'

type Props = {
  data: Object,
  rowIndex: Number,
  deleteDefectFromId: Function,
  deleteDefectFromIndex: Function,
  saveDefect: Function,
  updateDefect: Function
};

export class ActionCellRender extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.onDelete = this.onDelete.bind(this)
    this.onSave = this.onSave.bind(this)
    // this.state = {
    //   isDeleteBtnOpen: false
    // }
  }

  onDelete () {
    if (this.props.data.id) {
      // 刪除後端資料庫的 defect
      this.props.deleteDefectFromId(this.props.data.id.toString())
    } else {
      // 刪除 redux state 裡的 defect list
      this.props.deleteDefectFromIndex(this.props.rowIndex)
    }
    // this.setState({
    //   isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    // })
  }

  onSave () {
    if (this.props.data.id) {
      // 如果有 id => 修改 defect (PUT)
      this.props.updateDefect(this.props.data)
    } else {
      // 如果沒有 id => 儲存 defect (POST)
      this.props.saveDefect(this.props.data, this.props.rowIndex)
    }
  }

  render () {
    return (
      <div>
        <section>
          <Button className={'pt-button pt-minimal pt-icon-trash'} title='Delete' onClick={this.onDelete} />
          <Button className={'pt-button pt-minimal pt-icon-confirm'} title='Save' onClick={this.onSave} />
        </section>
      </div>
    )
  }
}

export default connect(
  (state) => ({}),
  (dispatch) => ({
    deleteDefectFromId: (delId) => dispatch(delLabelingToolImageDefect(delId)),
    deleteDefectFromIndex: (rowIndex) => dispatch({
      type: DELETE_DEFECT_MARK_FROM_INDEX,
      data: {rowIndex}
    }),
    saveDefect: (defectData, rowIndex) => dispatch(saveLabelingToolImageDefect(defectData, rowIndex)),
    updateDefect: (defectData) => dispatch(updateLabelingToolImageDefect(defectData))
  })
)(ActionCellRender)

// export default ActionCellRender
