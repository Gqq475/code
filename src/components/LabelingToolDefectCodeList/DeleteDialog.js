import React from 'react'
import classes from './Dialog.scss'
import { Button, Dialog } from '@blueprintjs/core'

type Props = {
  id: Number,
  isOpen: boolean,
  toggleDialog: Function
};

export class DeleteDialog extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit () {
    this.props.toggleDialog()
  }

  render () {
    return (
      <div className={classes['Dialog-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>提示</h6>
          </div>
          <div className={classes.dialogBody}>
            <br />
            <p className={classes.cn}>{`是否删除id为${this.props.id}的这条数据?`}</p>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                {_.CANCEL}
              </Button>
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                {_.OK}
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}

export default DeleteDialog
