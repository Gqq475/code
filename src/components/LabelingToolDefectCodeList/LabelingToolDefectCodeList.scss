.LabelingToolDefectCodeList-container {
  width: 39%;
	float: right;
	overflow: hidden;
	border: 1px solid #0087DC;
	border-radius: 5px;
	padding: 0 0 10px 0;
	height: 215px;
	.cn{
    text-align: center;
  }
}
