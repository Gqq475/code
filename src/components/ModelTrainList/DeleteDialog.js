import React from 'react'
import classes from './Dialog.scss'
import { Button, Dialog } from '@blueprintjs/core'
import OurToaster from 'components/OurToaster'
import { connect } from 'react-redux'
const qca = __QCA__

type Props = {
  id: Number,
  api: Object,
  isOpen: boolean,
  toggleDialog: Function,
  token: String
};

export class DeleteDialog extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.handleSubmit = this.handleSubmit.bind(this)
  }

  handleSubmit () {
    // const url = `${qca}/ojs/service`
    const url = `${qca}/offline/version_control/model_version`
    const delData = {}
    delData.model_id = this.props.id.toString()
    fetch(url, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        'Accept': 'application/json',
        'Authorization': this.props.token
      },
      body: JSON.stringify(delData)
    })
    .then(response => response.json())
    .then(data => {
      if (data.code === 200000) {
        this.props.api.removeItems(this.props.api.getSelectedNodes())
      } else {
        OurToaster.show({message: '删除失败！'})
      }
    })

    this.props.toggleDialog()
  }

  render () {
    return (
      <div className={classes['Dialog-container']}>
        <Dialog
          isOpen={this.props.isOpen}
          onClose={this.props.toggleDialog}
          className={classes.dialog}
        >
          <div className={classes.header}>
            <h6 className={classes.headerTitle}>提示</h6>
          </div>
          <div className={classes.dialogBody}>
            <br />
            <p className={classes.cn}>{`是否删除id为${this.props.id}的这条数据?`}</p>
            <div className={classes.btnGroup}>
              <Button className={classes.btnCancel} onClick={this.props.toggleDialog}>
                {_.CANCEL}
              </Button>
              <Button className={`pt-button pt-intent-primary ${classes.btnEdit}`} onClick={this.handleSubmit}>
                {_.OK}
              </Button>
            </div>
          </div>
        </Dialog>
      </div>
    )
  }
}
export default connect(
  (state) => ({
    token: state.auth.token
  }),
  (dispatch, state) => {
    return {
      dispatch
    }
  }
)(DeleteDialog)
