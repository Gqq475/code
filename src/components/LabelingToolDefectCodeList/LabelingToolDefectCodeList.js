import React from 'react'
import classes from './LabelingToolDefectCodeList.scss'
import {AgGridReact} from 'ag-grid-react'
import ActionCellRender from './ActionCellRender'
// import Utils from '../../Utils.js'

type Props = {
  defectList: Array,
  selectDefect: Function
};

export class LabelingToolDefectCodeList extends React.Component {
  props: Props;
  constructor (props) {
    super(props)
    this.onRowClicked = this.onRowClicked.bind(this)
    const columnDefs = [
      {
        headerName: 'No',
        width: 60,
        field: 'id',
        suppressMovable: true,
        cellClass: classes.cn
      }, {
        headerName: 'Defect Code',
        field: 'name',
        suppressSorting: true,
        suppressMovable: true,
        cellClass: classes.cn
      }, {
        headerName: 'Image Id',
        width: 90,
        field: 'imageId',
        suppressMovable: true,
        cellClass: classes.cn
      }, {
        headerName: 'Action',
        suppressMovable: true,
        width: 90,
        suppressSorting: true,
        cellRendererFramework: ActionCellRender
      }
    ]

    this.state = {
      columnDefs
    }
  }

  // datasource () {
  //   return {
  //     rowCount: 1,
  //     getRows: (params) => {
  //       let pageSize = 10
  //       let pageNo = params.endRow / pageSize
  //       const url = Utils.format(this.props.data.labelingTem, [pageNo, pageSize])
  //       fetch(url)
  //       .then(response => response.json())
  //       .then((data) => {
  //         console.dir(data.result.imageDefects)
  //         params.successCallback(data.result.imageDefects)
  //       })
  //     }
  //   }
  // }
  onRowClicked (rowInfo) {
    this.props.selectDefect(rowInfo.rowIndex)
  }

  render () {
    return (
      <div className={`${classes['LabelingToolDefectCodeList-container']} ag-fresh`}>
        <AgGridReact
          onRowClicked={this.onRowClicked}
          rowSelection={'single'}
          columnDefs={this.state.columnDefs}
          rowData={this.props.defectList}
          rowHeight={30}
        />
      </div>
    )
  }
}

export default LabelingToolDefectCodeList
