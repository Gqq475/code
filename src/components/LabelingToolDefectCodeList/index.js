import LabelingToolDefectCodeList from './LabelingToolDefectCodeList'
export default LabelingToolDefectCodeList
