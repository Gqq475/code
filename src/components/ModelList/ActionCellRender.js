import React from 'react'
import DeleteDialog from './DeleteDialog'
import classes from './ModelList.scss'

type Props = {
  data: Object,
  api: Object
};

export class ActionCellRender extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.onDelete = this.onDelete.bind(this)
    this.state = {
      isDeleteBtnOpen: false
    }
  }

  onDelete () {
    this.setState({
      isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    })
  }

  render () {
    return (
      <div className={classes.cn}>
        <button onClick={this.onDelete}
          className={'pt-button pt-minimal pt-icon-delete'}
          title='Delete' />
        <DeleteDialog
          api={this.props.api}
          id={this.props.data.id}
          isOpen={this.state.isDeleteBtnOpen}
          toggleDialog={this.onDelete} />
      </div>
    )
  }
}

export default ActionCellRender
