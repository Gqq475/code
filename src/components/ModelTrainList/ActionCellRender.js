import React from 'react'
// import AddDialog from './AddDialog'
import DeleteDialog from './DeleteDialog'
// import PublishDialog from './PublishDialog'
// import ModifyDialog from './ModifyDialog'
import classes from './ModelTrainList.scss'
// import OurToaster from 'components/OurToaster'

type Props = {
  data: Object,
  api: Object
};

export class ActionCellRender extends React.Component {
  props: Props;

  constructor (props) {
    super(props)
    this.onDelete = this.onDelete.bind(this)
    this.state = {
      isDeleteBtnOpen: false
    }
  }

  onDelete () {
    this.setState({
      isDeleteBtnOpen: !this.state.isDeleteBtnOpen
    })
  }

  render () {
    return (
      <div className={classes.cn}>
        <button onClick={this.onDelete}
          className={'pt-button pt-minimal pt-icon-delete'}
          title='Delete' />
        <DeleteDialog
          api={this.props.api}
          id={this.props.data.id}
          isOpen={this.state.isDeleteBtnOpen}
          toggleDialog={this.onDelete} />
      </div>
    )
  }
}

export default ActionCellRender
