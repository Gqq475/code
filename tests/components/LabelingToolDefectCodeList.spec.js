import React from 'react'
import LabelingToolDefectCodeList from 'components/LabelingToolDefectCodeList/LabelingToolDefectCodeList'
import { ActionCellRender } from 'components/LabelingToolDefectCodeList/ActionCellRender'
import { DeleteDialog } from 'components/LabelingToolDefectCodeList/DeleteDialog'
describe('(Component) LabelingToolDefectCodeList', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <LabelingToolDefectCodeList />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
describe('(Component) ActionCellRender', () => {
  it('render normally', () => {
    const data = {
      'id': '1'
    }
    const wrapper = shallow(
      <ActionCellRender data={data} />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
describe('(Component) DeleteDialog', () => {
  it('render normally', () => {
    const wrapper = shallow(
      <DeleteDialog />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
