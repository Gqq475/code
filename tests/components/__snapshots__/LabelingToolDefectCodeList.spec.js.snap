// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) ActionCellRender render normally 1`] = `
<div>
  <section>
    <Blueprint.Button
      className="pt-button pt-minimal pt-icon-trash"
      onClick={[Function]}
      title="Delete"
    />
    <Blueprint.Button
      className="pt-button pt-minimal pt-icon-confirm"
      onClick={[Function]}
      title="Save"
    />
  </section>
</div>
`;

exports[`(Component) DeleteDialog render normally 1`] = `
<div
  className="Dialog-container"
>
  <Dialog
    className="dialog"
    isOpen={false}
  >
    <div
      className="header"
    >
      <h6
        className="headerTitle"
      >
        提示
      </h6>
    </div>
    <div
      className="dialogBody"
    >
      <br />
      <p
        className="cn"
      >
        是否删除id为undefined的这条数据?
      </p>
      <div
        className="btnGroup"
      >
        <Blueprint.Button
          className="btnCancel"
        />
        <Blueprint.Button
          className="pt-button pt-intent-primary btnEdit"
          onClick={[Function]}
        />
      </div>
    </div>
  </Dialog>
</div>
`;

exports[`(Component) LabelingToolDefectCodeList render normally 1`] = `
<div
  className="LabelingToolDefectCodeList-container ag-fresh"
>
  <Component
    columnDefs={
      Array [
        Object {
          "cellClass": "cn",
          "field": "id",
          "headerName": "No",
          "suppressMovable": true,
          "width": 60,
        },
        Object {
          "cellClass": "cn",
          "field": "name",
          "headerName": "Defect Code",
          "suppressMovable": true,
          "suppressSorting": true,
        },
        Object {
          "cellClass": "cn",
          "field": "imageId",
          "headerName": "Image Id",
          "suppressMovable": true,
          "width": 90,
        },
        Object {
          "cellRendererFramework": [Function],
          "headerName": "Action",
          "suppressMovable": true,
          "suppressSorting": true,
          "width": 90,
        },
      ]
    }
    onRowClicked={[Function]}
    rowHeight={30}
    rowSelection="single"
  />
</div>
`;
